### ISSUE TITLE
TermObject.uri strips home_url() after term_link filter has been applied

### ISSUE THREAD

**Issue by jasonbahl** [wp-graphql/wp-graphql#2210]
This issue is copied from a Slack conversation with @theodesp

-----

I have a question related to some investigation I had to do regarding Term uri’s.

Regarding the following query:

```graphql
query MyQuery {
  terms {
    nodes {
      uri
      link
    }
  }
}
```

This returns:

```json
"data": {
    "terms": {
      "nodes": [
        {
          "uri": "/posts/category/uncategorized/",
          "link": "http://localhost:10003/posts/category/uncategorized/"
        }
      ]
    }
  },
  ```
  
I noticed that for computing the `uri` part WPGraphQL takes the term links in absolute URI from the TermObject here:
https://github.com/wp-graphql/wp-graphql/blob/7cab4f0184fbc5b2eb0a06ba3c518c896cf6e26d/src/Type/ObjectType/TermObject.php#L55
and converts it to relative path stripping the `home_url()`.

Someone would expect it to call the term `uri` from the Term model instead using this case here:
https://github.com/wp-graphql/wp-graphql/blob/7cab4f0184fbc5b2eb0a06ba3c518c896cf6e26d/src/Model/Term.php#L196-L205

The thing is with the above call to `TermObject.php#L55` it triggers subsequently the link  section instead as I was walking through the debugger here:
https://github.com/wp-graphql/wp-graphql/blob/7cab4f0184fbc5b2eb0a06ba3c518c896cf6e26d/src/Model/Term.php#L166-L169

Is there a reason that it goes through that flow?

I ask because in Faust.js we use a term link filter to replace the site_url with the headless site url

We expect an absolute link and we return an absolute link as normal:
https://github.com/wpengine/faustjs/blob/c74ce4f80f933ab73ffed43697d32e35d0b40311/plugins/faustwp/includes/replacement/callbacks.php#L215-L228

but this causes this issue:
https://wpengine.atlassian.net/browse/FST-58

so whenever we call the `get_term_link()`  we want to get an absolute link and replace the `site_url` with the `frontend_url`

However now with the current flow the frontend_url  does not match the base_url so the replacement fails and it will return the absolute path instead:

```
"data": {
    "terms": {
      "nodes": [
        {
          "uri": "http://localhost:3000/posts/category/uncategorized/",
          "link": "http://localhost:3000/posts/category/uncategorized/"
        }
      ]
    }
  },
  ```

I was wondering if WPGraphQL could just make the uri resolver more generic and instead of replacing the home_url() we replace  the host:port path  which may not be the same if we have similar filters in place.

For example: `"http://www.example.com/example"` to `"/example"` even if `home_url()` is different than `http://www.example.com`

If this is too risky to do that, is there anyway we can control how the uri is resolved here?

**Comment by jasonbahl** [comment]
@theodesp thanks for opening this issue. 

I think your suggestion of replacing the host/port sounds like a good plan. 🙌🏻 

I think it would solve the same thing I was trying to solve, but would still give flexibility for use cases like you describe where the term_link has already filtered the domain before the uri resolver gets it.

**Comment by jasonbahl** [comment]
@theodesp would you be interested in submitting a PR to make this change and including / updating tests for it?

**Comment by theodesp** [comment]
@jasonbahl thank you. Will discuss it with the @faust.js team to see what we can do here.

**Comment by moonmeister** [comment]
Seems like the code I just used in #2182 is relevant here: https://github.com/wp-graphql/wp-graphql/pull/2182/files#diff-95fd90c4e82eb78455558373d51bfb443010152c93eb52b9ff9730cff2a80c96R185

**Comment by jasonbahl** [comment]
@moonmeister yes, good call. very similar, and likely an issue that arises for the same user, that's accessing the WordPress install from a different domain than site_url() is set to

**Comment by moonmeister** [comment]
Fyi, Jason the way FaustWP plugin currently assumes home and site url are the same and set to the backend. It provides it's one frontend url from a custom config. We just had a talk with @apmatthews and @blakewilson about changing this. Gatsby also does this

**Comment by jasonbahl** [comment]
I think it's a decent idea to keep a separate option, specifically for the time period users are building their headless site while their WordPress site is still live. 

But, I think there should also be an option for Faust/Gatsby, to "use site_url()" or whatever, for the cases where the front-end url is the same as the decoupled url.

**Comment by jasonbahl** [comment]
closed by #2212
